bip-internalkey.mediawiki

```diff
@@ -0,0 +1,70 @@
+<pre>
+  BIP: TBD
+  Layer: Consensus (soft fork)
+  Title: INTERNALKEY
+  Author: Brandon Black <freedom@reardencode.com>
+  Comments-Summary: No comments yet.
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-12-22
+  License: PD
+</pre>
+
+==Abstract==
+
+This BIP describes a new tapscript opcode (<code>OP_INTERNALKEY</code>) which
+pushes the taproot internal key to the stack.
+
+==Specification==
+
+When verifying taproot script spends having leaf version 0xc0 (as defined in
+BIP342), <code>OP_INTERNALKEY</code> replaces <code>OP_SUCCESS203</code>
+(0xcb). <code>OP_INTERNALKEY</code> pushes the taproot internal key, as defined
+in BIP341, to the stack.
+
+==Motivation==
+
+===Key spend with additional conditions===
+
+When building taproot outputs, especially those secured by an aggregate key
+representing more than 1 signer; the parties may wish to collaborate on signing
+with the taproot internal key, but only with additional script restrictions. In
+this case, <code>OP_INTERNALKEY</code> saves 32 bytes of witness space.
+
+===Mitigated control block overhead for scripts using hash locks===
+
+In cases where script path spending is not desired, the internal key may be set
+to a NUMS point whose bytes would otherwise be required in a tapscript. This
+could be used with any hash locked transaction, for example, to save 32 witness
+bytes.
```

Comment:
The internal key needs to correspond to a point on the secp256k1 curve, so how can this be used with "any hash locked transaction"?

Reply:
Yeah, I need to tighten up this language. In a typical case it will take about 2 tries to find a hash that is on the curve. For hash locks, a next preimage can be derived if the first does not hash to an X on the curve; for CTV locks, the template transaction can be modified in some slight protocol-compatible way to get a hash that is also an X on the curve.